.github/workflows/self-tests-macos.yaml

```diff
@@ -1,4 +1,4 @@
-name: Build and validate
+name: macOS self-tests
```

Comment:
I initially thought the same about the matrix, but I think it might be useful to keep it in to make it easy for us if we want to add testing for different versions of the same OS (e.g. Ubuntu 18.04 as well as Ubuntu 20.04). 

Unfortunately, I don't think it's possible to avoid the code duplication here. AFAIK, the YAML language can't perform such logic; it has no concept of "import" or "include".

Reply:
**Update:** Removed the matrix as mentioned.